cngi/image/reframe.py

```diff
@@ -15,22 +15,239 @@
+            )
+            # note that we could include telescope dependent conditionals (e.g., FK5 vs ICRS)
+            # these could likely be based on ixds.attrs["direction_reference"]
+        finally:
+            print(f"Adopted target location: {target_location}")
+            if target_location.is_transformable_to(outframe):
+                pass
+            else:
+                print("Input to outframe argument incompatible with target position")
+                print(
+                    "Adding velocity coordinate for per-channel assignment to target_location"
+                )
+                # if velocity information isn't present in the input
+                # (and it won't be because that logic hasn't been encoded due to available test data)
+                # then calculate a velocity for each channel and assign as dataset coordinate,
+                # for selection inside the mapped function (_change_frame)
+                measured_frequencies = _ixds.chan.values * units.Unit(
+                    _ixds.attrs["units_map"]["Frequency"]
+                )
+                radio_equiv = units.doppler_radio(reference_frequency)
+                measured_velocities = measured_frequencies.to(
+                    units.km / units.s, equivalencies=radio_equiv
+                )
+                _ixds = txds.assign_coords(velocity=("chan", measured_velocities.value))
+                _ixds.attrs["units_map"][
+                    "Velocity"
+                ] = measured_velocities.unit.to_string()
+
+    # Defining a function to apply the transformation across the chan axis
+    def _change_frame(ixds_block, observer, target, frequency, frame):
+        """
+        This function will be called using xr.map_blocks
+
+        reference
+        http://xarray.pydata.org/en/stable/generated/xarray.map_blocks.html
+        and
+        https://xarray-contrib.github.io/xarray-tutorial/scipy-tutorial/06_xarray_and_dask.html#map_blocks
+        """
+
+        if ("velocity" in ixds_block.coords) and (
+            "Velocity" in ixds_block.attrs["units_map"]
+        ):
+            # update target to use velo coordinate value for this block
+            # since this is inside map_blocks, the only velo/chan available is the one we want to use
+            chan_vel = ixds_block.velocity.values * units.Unit(
+                ixds_block.attrs["units_map"]["Velocity"]
+            )
+            # assignment via `SkyCoord(target, radial_velocity=chan_vel)` doesn't seem to work
+            # see 
+            # https://docs.astropy.org/en/stable/coordinates/velocities.html#adding-velocities-to-existing-frame-objects
+            # https://docs.astropy.org/en/stable/coordinates/representations.html#attaching-differential-objects-to-representation-objects
+            target = target.data.with_differentials({'s':RadialDifferential(chan_vel)})
```

Comment:
something about applying the calculated velocities to the SkyCoord object in this way still does not work right for all tested `outframe` inputs